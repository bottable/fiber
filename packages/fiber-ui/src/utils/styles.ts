import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
