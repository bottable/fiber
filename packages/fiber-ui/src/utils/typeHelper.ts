export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
