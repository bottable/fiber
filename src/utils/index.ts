export * from './typeHelper'
export * from './mergeProps'
