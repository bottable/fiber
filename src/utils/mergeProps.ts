import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
