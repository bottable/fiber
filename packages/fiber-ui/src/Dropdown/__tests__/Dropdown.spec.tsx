import { Dropdown, Menu, Text } from '../../'

import React from 'react'
import { render } from 'test/utils'

describe('Menu & Menu Items', () => {
  it('renders correctly', () => {
    const menu = (
      <Menu>
        <Menu.Item>
          <Text> Item 1 </Text>
        </Menu.Item>
        <Menu.Item>
          <Text> Item 2 </Text>
        </Menu.Item>
        <Menu.Item>
          <Text> Item 3 </Text>
        </Menu.Item>
      </Menu>
    )

    const dropdown = render(
      <Dropdown overlay={menu} trigger='hover'>
        <Text> Hover Me </Text>
      </Dropdown>
    )

    expect(dropdown).toMatchSnapshot()
  })
})
