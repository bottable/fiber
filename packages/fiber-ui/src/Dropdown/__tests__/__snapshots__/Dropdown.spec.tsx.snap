// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu & Menu Items renders correctly 1`] = `
<div
  className="sc-fzoiQi cGtddt"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="sc-fzonZV ckbpRN"
    size={4}
  >
     Hover Me 
  </span>
  <div
    className="sc-fzqARJ HtEMr"
  >
    <ul
      className="sc-fznZeY hBvHBb"
    >
      <li
        className="sc-fzokOt kZMlVH"
        onClick={[Function]}
      >
        <span
          className="sc-fzonZV ckbpRN"
          size={4}
        >
           Item 1 
        </span>
      </li>
      <li
        className="sc-fzokOt kZMlVH"
        onClick={[Function]}
      >
        <span
          className="sc-fzonZV ckbpRN"
          size={4}
        >
           Item 2 
        </span>
      </li>
      <li
        className="sc-fzokOt kZMlVH"
        onClick={[Function]}
      >
        <span
          className="sc-fzonZV ckbpRN"
          size={4}
        >
           Item 3 
        </span>
      </li>
    </ul>
  </div>
</div>
`;
